### ISSUE TITLE
Who is responsible for the user interface for PiP?

### ISSUE THREAD

**Issue by becka11y** [w3c/picture-in-picture#179]
As a representative of the Accessibility Platform Architectures (APA) working group I reviewed this specification for accessibility concerns. One accessibility use case we see is the display of sign language translation within the picture in picture window. 

I am trying to understand who is responsible for generating the user interface to find and switch between picture in picture windows?  Is the browser responsible - similar to the controls provided for the video element?  Or, is the web developer responsible for including the user interface?  Either way, it is imperative that this interface meets the requirements of WCAG 2.1.  We are asking that you add an accessibility statement to the specification that outlines the requirements for meeting WCAG 2.1 and especially of the mechanisms for a user to find, enable/disable, and navigate to/from the picture in picture.

**Comment by beaufortfrancois** [comment]
Thanks for raising this issue @becka11y!
I'd be happy to address those in the spec. Do you have examples of other web APIs specifications that demonstrate how to meet the requirements of WCAG 2.1? Would you be able us reaching that goal?

FYI @mounirlamouri

**Comment by becka11y** [comment]
Apologies for the delay in responding. We don't have any specific examples from other Web API specifications but the APA working group agreed upon the following suggestion for an  accessibility statement within the specification: 

It is important that the interface for picture in picture meets the requirements of the Web Content Accessibility Guidelines (WCAG). All users must be able to find and navigate to/from the picture in picture and enable/disable/operate all controls using various input modalities, color and contrast settings, and assistive technologies.

For example, the WCAG Guideline 2.1 requires, "Make all functionality available from a keyboard". Any picture in picture implementation and interface must be full operational with the keyboard only.

**Comment by tomByrer** [comment]
> must be full operational with the keyboard only

So will the keyboard-focus automatically bounce to the first 'tab focus' inside the PiP when it appears?  @becka11y, it not seem you answered the original topic if the browser be responsible for that auto-focus, or the webmaster is to program somethin in JavaScript.?

**Comment by Rasubra8** [comment]
I also have A similar query. If caption is not displayed in PIP mode is that A vialation of WCAGSC 1.2.2?

**Comment by chrisn** [comment]
@Rasubra8 Caption display in PiP windows is a recognised issue, where some browsers support rendering of captions in the PiP window, and others do not, and in some cases captions are rendered by the web application. But, yes, the PiP window should render captions.

**Comment by theIDinside** [comment]
It looks like PIP at some point was intended to have its UI (controls, etc) be configured by how the Media session API has been set up. It's currently not in spec but chrome seem to have this behavior @beaufortfrancois ?

Exploring the idea, it seems like this could be a programmatic way to make sure different user agents behave consistently, even if their UI's have a different look and feel, for instance. I don't know what it would do with respect to accessibility, though.

EDIT: I've found some pretty reasonable examples here https://googlechrome.github.io/samples/media-session/video-conferencing.html (one has to use Chrome to try this. Neither Firefox nightly PIP implementation nor Safari supports it).

**Comment by beaufortfrancois** [comment]
https://web.dev/articles/media-session is a good summary of how the Media Session API impacts UI, including PiP.

I'm not sure we'd want to spec this behavior though as the PiP window is just yet another surface where the different media session actions can live.

**Comment by theIDinside** [comment]
> https://web.dev/articles/media-session is a good summary of how the Media Session API impacts UI, including PiP.
> 
> I'm not sure we'd want to spec this behavior though as the PiP window is just yet another surface where the different media session actions can live.

Yeah, but I don't see how the spec suggests this should happen. If we had some spec language to suggest how this should be done, I think both Safari and Firefox could plan for how to support this. @marcoscaceres do you know what's the status on this in Safari?

**Comment by marcoscaceres** [comment]
Had a look across browsers and implementations... this is where I landed https://github.com/w3c/picture-in-picture/pull/264
